mopro-ffi/src/app_config/mod.rs

```diff
@@ -1,73 +1,74 @@
-use std::fs;
-use std::path::{Path, PathBuf};
-use std::process::Command;
-use uuid::Uuid;
+use camino::Utf8Path;
+use uniffi_bindgen::bindings::{KotlinBindingGenerator, SwiftBindingGenerator};
+use uniffi_bindgen::library_mode::generate_bindings;
 
-pub mod android;
-pub mod ios;
+use crate::app_config::utils::build_release;
 
-pub fn mktemp() -> PathBuf {
-    let dir = std::env::temp_dir().join(Path::new(&Uuid::new_v4().to_string()));
-    fs::create_dir(&dir).expect("Failed to create tmpdir");
-    dir
-}
+mod android;
+mod ios;
+mod utils;
 
-fn tmp_local(build_path: &Path) -> PathBuf {
-    let tmp_path = build_path.join("tmp");
-    if let Ok(metadata) = fs::metadata(&tmp_path) {
-        if !metadata.is_dir() {
-            panic!("non-directory tmp");
-        }
-    } else {
-        fs::create_dir_all(&tmp_path).expect("Failed to create local tmpdir");
-    }
-    tmp_path
-}
+use utils::*;
 
-pub fn mktemp_local(build_path: &Path) -> PathBuf {
-    let dir = tmp_local(build_path).join(&Uuid::new_v4().to_string());
-    fs::create_dir(&dir).expect("Failed to create tmpdir");
-    dir
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum Target {
```

Comment:
I was thinking to unify the build for adnroid and ios by eventually providing a single binary instead of two, and using a march statement there to choose the architecture. 

Generally, I found that some of the logic was shared, and thought to use this enum to indicate which path to take for simplicity.

Do you find that it will be better to have two (or more later) different functions, with a sub functoin to call for shared logic? If so, I am happy to revert to the original design.

Reply:
This seems like premature optimization. If we do combine the logic in the future we can always add the enum or figure out a different solution then. For now having two different functions is a fine solution and doesn't restrict our options for the future.